### ISSUE TITLE
Pyro Rework ideas

### ISSUE THREAD

**Issue by fpsmoto** [fortressforever/fortressforever#224]
It's pretty clear that since FF's release, Pyro has been the class that most people do not like and I'd like to try to provide some suggestions for how to improve Pyro in the future.

**Being on fire**
There's nothing quite like lighting someone on fire nearby and hearing that shrieking lvl 3 scream. I think that's a good feature to have and could use a different scream for each burn level rather than just the one at level 3.

Another thing has to do with the automatic level 3 burn the moment you use each type of flame weapon. It should work so you can use even just 1 flame weapon to reach a level 3 burn, depending on how long they are exposed to the flames.

To accomplish this, a percentage-based system would be ideal.
- Each flame weapon would have a default % value for each hit with the weapon. (eg 5% for flamethrower per tick, 10% for each IC shot)
- Create a burn-level threshold so once you reach a certain percentage, it'd bump you up to a level 2 or 3 burn.
- Have this % decrease over time, allowing a more natural way to get rid of afterburn. Say lvl 3 burn requires 75%. If you get hit with flame weapons that go above that threshold to, lets say, 78%, you'd only be burning at the level 3 for a short period of time as that percentage would decrease down below that threshold back to level 2 burn, provided you stopped being hit with flame weapons during that time.

**The Flamethrower**
Pyro should have to slow down while using the Flamethrower, similar in concept to how the HW Guy slows while using his Assault Cannon.
The current hit box of the flamethrower is kind of ridiculous. After seeing an image of the hitbox, it extends way beyond the visible flames and it's a rectangle when it should be more of a cone shape. Right now, the hitbox width is roughly the size of the biggest part of the flames. This makes no sense for the smallest part that's coming out of the flamethrower and probably results in people walking near a Pyro using a flamethrower and inadvertently getting hit with it, even if the Pyro isn't aiming directly at them.

**Afterburn**
Use the same % based system to apply to speed, so the faster you are going, the faster the Afterburn dissipates. Any type of speed boost you get from trimping to jump pads to nade jumping would have this potential benefit.

**Comment by Chickenprotector** [comment]
Speed DOES NOT need to be nerfed when using it, it defeats the purpose of how the pyro's supposed to play, on offense or on defense. The primary role of a pyro is DPS. Plain and simple. In other words, clearing out the enemy be it a boat load of scouts, medics, or such on offense or eliminating an annoying sentry gun for the scouts/medics on your team.

I like the tick damage re-work but DPS is still an issue with this, FT still does 80/sec ignoring afterburn, and the hitbox should be cut significantly, that much I agree with too.

To put it into something understandable, the system would work similarly to the Medic's infection currently works. The first tick does 70 damage and then reduces as it goes (being affected by things like speed and/or water). There should be a speed threshold before it reduces the tick percentage. You shouldn't be able to reduce the percentage unless you've been conced, etc. (700+ sounds reasonable).

Armament is also an issue, the IC is not a reliable weapon to use to finish someone off.

**Comment by fpsmoto** [comment]
The problem is Pyro is as fast as 4 out of 5 FF classes, so when hit by a flamethrower, a lot of the time it's impossible to escape. With the speed decrease while using it, you'd still be able to alternate between bhopping and firing the flamethrower to maintain some of your speed. It's either that or the pyro needs an overall speed nerf of something like 270-280 compared to his default speed of 300.

What say you about the HW using his fart special to put out fire as well for himself and/or nearby teammates?

**Comment by FDA-** [comment]
Remove afterburn entirely. Cross post from https://github.com/fortressforever/fortressforever/issues/216

"I think after burn is a largely undeserved reward for a relatively easy application, and I don't really think DoT type mechanics really belong in an FPS. It's just not a fun interaction. I can think of some ways of molding the pyro into a more interesting DM system, but possibly at the cost of the pub experience?

One thing I thought of was making afterburn a "damage modifier" tag, rather than straight up damage. So say you hit someone with your flame thrower (the easier to hit weapon), then the afterburn tag is applied which makes your IC deal more damage. (the more skilled to hit with weapon?) In a system like that it'd probly be good to significantly reduce the afterburn duration, requiring more weapon juggling. That way it's not just a mindless automation of touch with flame thrower, spam with IC.

Alternatively I've also always thought it'd be interesting to make the pyro more "quake like". Significantly reduce the flame thrower cone of fire into a more concentrated LG type weapon. Increase IC rate of fire, and reduce projectile size to make it more like the plasma gun. Probably leave the projectile ark to give it some sort of variation over the SNG. And of course adjust flame thrower and IC damage values to better fit their new mechanics.

Either the "damage bonus afterburn", or the "sustained fire lg flame" could be potentially devastating to a slow moving/immobile HW guy, so there's that to consider too."

**Comment by FDA-** [comment]
Basically I don't think the pyro needs a buff or a nerf, it needs a unique reimagining. Adding some silly formula to afterburn and making the pyro stand still while he shits on people doesn't really fit that. You're still dealing with the same shitty "being on fire while annoying screams sound off around you" scenario.

**Comment by fpsmoto** [comment]
I don't think afterburn is undeserved all the time. Sure, some times you are on offense and brush right past a pyro as they attack but they can't really catch up to you and you feel like you've just wasted a run into the base. Plus if they pull off a quick IC shot as you make your narrow escape from a painful death, you don't really stand a good chance to do anything productive from that point. And annoying as that can be, I don't think it should be removed.

Adding the ability for that afterburn damage to be reduced over time, plus go from a particular level burn down one or two levels while not actively being hit with a flame weapon would allow for afterburn to become more balanced without having to remove it. Couple that with the ability to get rid of the afterburn via speed and outside close range combat, the _afterburn_ would become an _afterthought_. This would significantly reduce deaths from afterburn and do less damage overall, but not to the point where it becomes useless.

Afterburn is a nice effect, adds some realism to the fold and it'd suck to see it be removed altogether. There are other DoT weapons that work well in FF. Why can't afterburn be one of them?

The issue with pyro's speed is more related to direct flame damage, not the afterburn. Since most classes are the same speed or slower than the pyro, the flamethrower becomes a death sentence for anyone who's within range aside from maybe soldier, scout or hw. This would be a simple solution that would take 0 coding and wouldn't take away from the pyros ability to murder people who cross their path, just allow for you to be able to escape their flamethrower a bit easier. Plus, this slowdown would only occur while using the flamethrower. They could still catch up to you and lay on the burn, but it gives a quick out for those being attacked by it.

I like your concept of a plasma gun type weapon for the IC though.

**Comment by ddm999** [comment]
I agree with having a percentage-based system and decreasing the flamethrower hitbox, but don't agree with slowing a pyro using the flamethrower or speed reducing / removing afterburn.
(I do agree with a slight nerf to his run speed though, which would also reduce his max speed from the bhop cap.)

As I see it, Pyro should be a defensive class. He should act somewhere in-between a Solly or a HW, capable of playing in both the HW's position and the Solly's position, but not **better** than these classes in their positions. His strength should be his ability to quickly rotate with flamethrower push to cover gaps in the defense caused by teammates dying, or even just as a backup 2-minutes-left-in-round switch to in comp. play for when things are going badly.

Making his speed reduced when using flamethrower just makes him an alternative / replacement for the HW.
Making the afterburn die out with speed means he has to play in the Solly's spot (what's the point in playing in the HW's position when the burn does barely any damage because everyone's rampsliding past you), and be an alternative / replacement for Solly instead.

I also agree with having afterburn placing a damage modifier for fire weapons.
I think best would be reducing the damage of the flamethrower, IC and napalm overall, and then having afterburn work percentage-based as Moto suggests; level 2 being 40% and level 3 being 80%, but also adding this percentage to the damage dealt by the flamethrower, IC and napalm.
(And also reducing the push from IC rockets, so Pyro's are slower - even if it's just a bit - to get to like 2000 u/s or whatever.)

**Comment by ph1sh55** [comment]
I disagree that pyro should be a defensive class.  Demoman, soldier, hwguy, engy are already solidly defensive- there's really no way you are going to displace any of those classes without significant overlap, and offense already suffers from lack of class diversity.  From the get go pyro was designed to be more offensive in FF- IMO he can have an offensive niche (ideally offensive DPS) that he can excel at.  He's close to viable, but has some a specific glaring weakness.  Right now at the top level of play offense is basically 2-3 medics, 1-2 scouts on every map, and an occasional spy being used every so often.  

Medics are superior movement wise (can get to the enemy base faster, conc, etc), are superior at taking down sentry guns w/ the nailgun (which is the biggest necessity on offense to give your scouts room to work) and on a round basis are likely on par or superior in pure DM ability.  (more runs = more damage dealt over the course of the round).  

Pyro on the other hand is completely inferior at taking down sentry guns which IMO is the main thing holding him back from being used on offense, and he requires more time to do his damage in most cases.  Mortars are very weak vs sentries and he can't close the distance on a SG as effectively as a medic.  When he does, his flamethrower is even inferior DPS wise vs sentries at moderately close range.

I would consider a few tweaks: 
- tighten up his flamethrower 'hit' range- make it require a bit more precision to deal optimal damage. Almost like a stream or focused jet of fire instead of a wide billowing flame as it is now.  
- Possibly introduce a charged shot from the mortar/IC to increase burst damage, or increase flame damage specifically against sentries.

IMO napalm on a sentry should be a death sentence for a gun much like a mirv.

**Comment by fpsmoto** [comment]
Eh, I've never really seen the Pyro as an offensive class. Like, not at all. I'd much rather see the Spy take that role and the Pyro go in the other direction. The Pyro's flamethrower used as a jetpack is novelty at best and he has no real other methods for which to be a viable option as an offensive class. He has heavy armor and a lot of it, is mainly a close-range class, good for choke points, and his weapons are not designed to do a lot of damage to highly armored classes or buildables in comparison to the medic or spy. So he's definitely more of a defensive class.

With that said, a charged IC shot would be cool as hell and I fully support that concept, but maybe with the added benefit of being able to shoot further or lower the amount of arc that happens due to the faster projectile.

**Comment by ph1sh55** [comment]
Well he currently can IC jump to improve speed in addition to the jetpack thing- but it would be kind of nice if the charged IC shot provided a bit more explosive push as well, so that he could do a more powerful jump (at the expense of more life and needing to 'prime' the shot) in certain situations.  Speeding up the projectile for a charged shot a touch/lowering the arc seems good - also would help against sentries long range.  He's never been a viable defensive class mostly because he has no stopping power- he can't really disrupt peoples movement.  Soldier, heavy, demo, engy all have stopping power one way or another and I think there's enough there.

On offense he wouldn't need stopping power, just DPS, and he's already naturally stronger against a heavy due the heavies lack of mobility and lack of burst.  His niche is probably best as a soft counter to heavy.  He'll still be naturally weaker vs soldiers or any mobile class who can bounce him around or get separation.
